Clarify TestRateLimit

douglas-reid opened the pull request:
This PR cleans up some of the bits of how the rate limit is evaluated and tested.

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note-none
NONE
```

istio-testing commented:
@douglas-reid: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-no_auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/214/) | `/test e2e-suite-no_rbac-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

ldemailly commented:
/lgtm

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-no_auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/214/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/214/) | `/test e2e-suite-rbac-auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-no_auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/214/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/214/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-no_rbac-auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/214/) | `/test e2e-suite-no_rbac-auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-no_auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/214/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/214/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-no_rbac-auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/214/) | `/test e2e-suite-no_rbac-auth`
prow/istio-presubmit.sh | 9acaf833f7020b80189ebb953056f16c7c27d658 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/istio-presubmit/602/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

douglas-reid commented:
ptal. i switched `TestRateLimit` to use the v3-only rule (instead of trying to split between v2/v3) which should make the value inspection simpler. i also added promDump logging to all appropriate tests to make it easier to pinpoint failures better (on the mixer side).

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-no_auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/214/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/214/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-no_rbac-auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/214/) | `/test e2e-suite-no_rbac-auth`
prow/istio-presubmit.sh | 15946bf608f254f45bcf2e23c8dfdc86f47cb6bb | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/istio-presubmit/603/) | `/test istio-presubmit`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

ldemailly commented on tests/e2e/tests/mixer/mixer_test.go:
```diff
@@ -163,16 +163,17 @@ func TestGlobalCheckAndReport(t *testing.T) {
 		t.Fatalf("Test app setup failure: %v", err)
 	}
+	// must sleep to allow for prometheus scraping, etc.
+	time.Sleep(30 * time.Second)
```
nit: maybe log before sleep is better to know the pause is normal

ldemailly commented on tests/e2e/tests/mixer/mixer_test.go:
```diff
@@ -202,21 +205,22 @@ func TestNewMetrics(t *testing.T) {
 	// allow time for configuration to go and be active.
 	// TODO: figure out a better way to confirm rule active
-	time.Sleep(5 * time.Second)
+	time.Sleep(30 * time.Second)
```
nit: that 30s could be a constant sleepTime or such

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-no_auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/214/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-no_rbac-auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/214/) | `/test e2e-suite-no_rbac-auth`
prow/e2e-suite-rbac-auth.sh | fe6ea7121bf296df4614488987866f46909bda28 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/227/) | `/test e2e-suite-rbac-auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-no_auth.sh | b7895f4080c6560f0cae10a9262f6c874e8b98ad | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/214/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | fe6ea7121bf296df4614488987866f46909bda28 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/227/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-no_rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/229/) | `/test e2e-suite-no_rbac-auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-rbac-auth.sh | fe6ea7121bf296df4614488987866f46909bda28 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/227/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-no_rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/229/) | `/test e2e-suite-no_rbac-auth`
prow/e2e-suite-no_rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/229/) | `/test e2e-suite-no_rbac-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/229/) | `/test e2e-suite-no_rbac-auth`
prow/e2e-suite-no_rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/229/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/229/) | `/test e2e-suite-rbac-auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/229/) | `/test e2e-suite-no_rbac-auth`
prow/e2e-suite-no_rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/229/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/229/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-no_auth/229/) | `/test e2e-suite-rbac-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

douglas-reid commented:
/retest

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/229/) | `/test e2e-suite-no_rbac-auth`
prow/e2e-suite-rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/229/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-no_auth/229/) | `/test e2e-suite-rbac-no_auth`
prow/e2e-suite-no_rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/231/) | `/test e2e-suite-no_rbac-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/229/) | `/test e2e-suite-no_rbac-auth`
prow/e2e-suite-rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-no_auth/229/) | `/test e2e-suite-rbac-no_auth`
prow/e2e-suite-no_rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/231/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/231/) | `/test e2e-suite-rbac-auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/229/) | `/test e2e-suite-no_rbac-auth`
prow/e2e-suite-no_rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/231/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/231/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-no_auth/231/) | `/test e2e-suite-rbac-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/231/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/231/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-no_auth/231/) | `/test e2e-suite-rbac-no_auth`
prow/e2e-suite-no_rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/231/) | `/test e2e-suite-no_rbac-auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

ldemailly commented on tests/e2e/tests/mixer/mixer_test.go:
```diff
@@ -95,7 +95,7 @@ func createDefaultRoutingRules() error {
 		return fmt.Errorf("could not create base routing rules: %v", err)
 	}
-	time.Sleep(30 * time.Second)
+	allowRuleSync()
```
👍

ldemailly commented on tests/e2e/tests/mixer/mixer_test.go:
```diff
@@ -344,33 +366,44 @@ func TestRateLimit(t *testing.T) {
+}
+
+func allowPrometheusSync() {
+	glog.Info("Sleeping to allow prometheus to record metrics...")
```
:+1:
(we could also have our own sleep that says how long it's about to sleep with optional "why" message)

douglas-reid commented:
I ran this twice in a row on my private cluster. Both times:

```
****************************************************
Tests Summary
PASSED: //tests/e2e/tests/mixer:go_default_test
PASSED: //tests/e2e/tests/bookinfo:go_default_test
```

douglas-reid commented:
I'll wait to see what `prow/e2e-suite-no_rbac-no_auth.sh` does, but either way, i'm merging unless there is an objection.

ldemailly commented:
/retest

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-no_auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/231/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/231/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-no_rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/231/) | `/test e2e-suite-no_rbac-auth`
prow/e2e-suite-rbac-no_auth.sh | e3a3cc1dc7c9a56b4b5402a5febdb9baa60e8f02 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-no_auth/256/) | `/test e2e-suite-rbac-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/231/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-no_rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/231/) | `/test e2e-suite-no_rbac-auth`
prow/e2e-suite-rbac-no_auth.sh | e3a3cc1dc7c9a56b4b5402a5febdb9baa60e8f02 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-no_auth/256/) | `/test e2e-suite-rbac-no_auth`
prow/e2e-suite-no_rbac-no_auth.sh | e3a3cc1dc7c9a56b4b5402a5febdb9baa60e8f02 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/256/) | `/test e2e-suite-no_rbac-no_auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-no_rbac-auth.sh | 2bc6c10844804d6492648dea87c46178aa879cf4 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/231/) | `/test e2e-suite-no_rbac-auth`
prow/e2e-suite-rbac-no_auth.sh | e3a3cc1dc7c9a56b4b5402a5febdb9baa60e8f02 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-no_auth/256/) | `/test e2e-suite-rbac-no_auth`
prow/e2e-suite-no_rbac-no_auth.sh | e3a3cc1dc7c9a56b4b5402a5febdb9baa60e8f02 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/256/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | e3a3cc1dc7c9a56b4b5402a5febdb9baa60e8f02 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/256/) | `/test e2e-suite-rbac-auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->

istio-testing commented:
@douglas-reid: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
prow/e2e-suite-rbac-no_auth.sh | e3a3cc1dc7c9a56b4b5402a5febdb9baa60e8f02 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-no_auth/256/) | `/test e2e-suite-rbac-no_auth`
prow/e2e-suite-no_rbac-no_auth.sh | e3a3cc1dc7c9a56b4b5402a5febdb9baa60e8f02 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-no_auth/256/) | `/test e2e-suite-no_rbac-no_auth`
prow/e2e-suite-rbac-auth.sh | e3a3cc1dc7c9a56b4b5402a5febdb9baa60e8f02 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-rbac-auth/256/) | `/test e2e-suite-rbac-auth`
prow/e2e-suite-no_rbac-auth.sh | e3a3cc1dc7c9a56b4b5402a5febdb9baa60e8f02 | [link](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/649/e2e-suite-no_rbac-auth/256/) | `/test e2e-suite-no_rbac-auth`



<details>

Instructions for interacting with me using PR comments are available [here](https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://github.com/kubernetes/test-infra/blob/master/commands.md).
</details>
<!-- test report -->
